modin/core/io/text/text_file_dispatcher.py

```diff
@@ -1014,9 +1014,18 @@ def _read(cls, filepath_or_buffer, **kwargs):
             names, skiprows, kwargs.get("skipfooter", 0), kwargs.get("usecols", None)
         )
 
+        skiprows = kwargs["skiprows"]
+        if names not in [None, lib.no_default] and skiprows is not None:
+            # The column names are already defined, there is no need to skip
+            # rows to find the right row to define the column names.
+            # We define this `skiprows` here as `1` instead of `None` to skip
+            # the header if there is one.
+            skiprows = 1
```

Comment:
why do we need to skip a header? the fact that we read the header shouldn't affect our metadata at all in case `names` are specified

Reply:
There is a tricky case here where the user can use `skiprows` parameter, `names` and `header!=0`. In this case, there will be an error here, since pandas will try to use the first row as the first column values, which will lead to type problems. Whereas the use of the `skiprows` parameter implicitly solves this problem for the user.

```bash
names: array-like, optional
    List of column names to use. If the file contains a header row, then you should explicitly pass header=0 to override the column names. Duplicates in this list are not allowed.
```

Reply:
Well, I'm afraid the problem you're trying to fix with `skiprows=1` is not fixed after all.

I've prepared a simple reproducer for the problem you're trying to avoid here (pay attention that I've specified `header=0` but still got the error):
```python
import modin.pandas as pd
import tempfile


with tempfile.NamedTemporaryFile() as file:
    pd.DataFrame(
        {"col1": ["str1", "str2", 1, 2, 3], "col2": ["str3", "str4", 5.2, 6.5, 7.7]}
    ).to_csv(file.name)

    df = pd.read_csv(
        file.name,
        skiprows=3,
        header=0,
        names=["col1", "col2"],
    ) # works fine

    df = pd.read_csv(
        file.name,
        skiprows=3,
        header=0,
        names=["col1", "col2"],
        dtype={"col1": int, "col2": float}
    ) # fails

```
Here I have a CSV file whose first half contains strings and the second half is purely numeric. What I want to do here is to discard the non-numeric part using `skiprows` parameter.

The error occurs because you explicitly pass numeric types here to read the `pd_df_metadata` that may not match actual types because of the modified `skiprows` parameter.

Reply:
@dchigarev I made a new implementation, ready for review.